### PR TITLE
Delete

### PR DESCRIPTION
**Pull request by asthaaaa**
Added methods for filename encoding
Added delete servlet for deleting indices
Added unit tests

### CODE REVIEW

File: src/main/java/com/google/error_analyzer/backend/IndexName.java [hunk 1]
```diff
@@ -52,10 +67,24 @@ private static String getSessionId(HttpServletRequest request) throws NullPointe
...
+        return sessionID;
     }
 
+    /*returns decoded index name*/
+    private static String decodeIndexName(String indexName) throws NumberFormatException {
+        String decodedIndexName = "";
+        String decimalStrings[] = indexName.split(STUFF_STRING);
+        for (String decimalString: decimalStrings) {
+            if(!decimalString.isEmpty()){
+            int decimalValue = Integer.parseInt(decimalString,16);
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/IndexName.java:82]: indentation?

File: src/main/java/com/google/error_analyzer/backend/IndexName.java [hunk 2]
```diff
@@ -52,10 +67,24 @@ private static String getSessionId(HttpServletRequest request) throws NullPointe
...
+            char character = (char)decimalValue;
+            String characterString = Character.toString(character);
+            decodedIndexName = decodedIndexName.concat(characterString);
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/IndexName.java:85]: we should be able to directly append a character to string.

File: src/main/java/com/google/error_analyzer/backend/LogDao.java [hunk 3]
```diff
@@ -166,6 +168,15 @@ public String getJsonStringById (String fileName, String id)
         return getResponse.getSourceAsString();  
     }
 
+    //delete indices
+    @Override
+    public String deleteIndices(String sessionId) throws IOException {
+        String indexName = sessionId.concat("*");
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/LogDao.java:174]: what if sessionID = ""?

File: src/main/java/com/google/error_analyzer/data/constant/FileConstants.java [hunk 4]
```diff
@@ -14,4 +14,6 @@
 /** This class contains all the constants related to files*/
 public class FileConstants {
     public static final String TEXT_HTML_CONTENT_TYPE = "text/html";
+    public static final String APPLICATION_JSON_CONTENT_TYPE =
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/data/constant/FileConstants.java:17]: Already present in master. Please sync branch to master.

File: src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java [hunk 5]
```diff
@@ -85,7 +85,7 @@ public void checkAndStoreFileLogExceptionCase() {
     /*store the file logs maximum 5 lines in a single API call*/
     @Test
     public void storeFileAndUrlLogsForFile() throws IOException {
-        String fileName = "file1";
+        String fileName = "i66i69i6ci65i31";
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java:88]: Is this filename or indexname?
Similarly below.

File: src/main/java/com/google/error_analyzer/backend/LogDao.java [hunk 6]
```diff
@@ -166,6 +168,15 @@ public String getJsonStringById (String fileName, String id)
         return getResponse.getSourceAsString();  
     }
 
+    //delete indices
+    @Override
+    public String deleteIndices(String sessionId) throws IOException {
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/LogDao.java:173]: deleteIndices(String indexRegex)

File: src/main/java/com/google/error_analyzer/backend/IndexName.java [hunk 7]
```diff
@@ -41,9 +44,21 @@ public static String getFileName(HttpServletRequest request,
         return fileName;
     }
 
+    /*returns encoded index name*/
+    public static String encodeIndexName(String indexName) {
+        String encodedIndexName = "";
+        for(int stringIndex = 0; stringIndex < indexName.length(); stringIndex++){ 
+            int asciiValue = (int) indexName.charAt(stringIndex);
+            String hex = Integer.toHexString(asciiValue);
+            encodedIndexName = encodedIndexName.concat(STUFF_STRING);
```
[the remark this answers is not in the document]
Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/IndexName.java:53]: We are adding STUFF_STRING "i" after hexadecimal value of each character to separate characters

File: src/main/java/com/google/error_analyzer/backend/LogDao.java (hunk 3 above)
Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/LogDao.java]: I changed the default sessionID to "invalidSession" so if the cookies is null (exception case) session id will be "invalidSession", if user clicks the delete button , then all the indices with index name pattern "indexName*" would be deleted. sessionID will not be equal to "" in any case.

File: src/main/java/com/google/error_analyzer/data/constant/FileConstants.java (hunk 4 above)
Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/data/constant/FileConstants.java:17]: I have raised this PR to StoreUrlLogs branch, will sync while merging to master

File: src/main/java/com/google/error_analyzer/backend/LogDao.java (hunk 3 above)
Comment (shudhatmajain) [reply src/main/java/com/google/error_analyzer/backend/LogDao.java]: Verifying: Will all indices with index name pattern "invalidSession*" be deleted (and not "indexName*")?

Approach sounds good to me.

Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/LogDao.java]: Sorry, indices with index name pattern  "invalidSession*" will be deleted

File: src/main/java/com/google/error_analyzer/backend/IndexName.java [hunk 8]
```diff
@@ -41,9 +44,21 @@ public static String getFileName(HttpServletRequest request,
         return fileName;
     }
 
+    /*returns encoded index name*/
+    public static String encodeIndexName(String indexName) {
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/IndexName.java:48]: Can we use a library like : 
import org.apache.commons.codec.binary.Hex

File: src/main/java/com/google/error_analyzer/backend/IndexName.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (shudhatmajain) [reply src/main/java/com/google/error_analyzer/backend/IndexName.java:53]: Can we rename it to "HEX_DELIMITER" or something on these lines.

File: src/test/java/com/google/error_analyzer/backend/StoreLogTest.java [hunk 9]
```diff
@@ -79,7 +79,7 @@ public void checkAndStoreLogInAlreadyExistingFileTest()
     @Test
     public void storeLogTestWithZeroOffset () throws IOException {
         String log = "error1\nerror2\nerror3";
-        String fileName = "file1";
+        String fileName = "i66i69i6ci65i31";
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/StoreLogTest.java:82]: indexName

File: src/test/java/com/google/error_analyzer/backend/StoreLogTest.java [hunk 10]
```diff
@@ -101,7 +101,7 @@ public void storeLogTestWithZeroOffset () throws IOException {
     @Test
     public void storeLogTestWithNonZeroOffset () throws IOException {
         String log = "error1\nerror2\nerror3";
-        String fileName = "file1";
+        String fileName = "i66i69i6ci65i31";
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/StoreLogTest.java:104]: indexName

File: src/test/java/com/google/error_analyzer/backend/IndexNameTest.java [hunk 11]
```diff
@@ -62,9 +62,18 @@ public void getIndexNameTest() {
...
+        String indexName = "i61i62i63i64i66i69i6ci65i31";
         String expected = "file1";
         String actual = IndexName.getFileName(request,indexName);
         Assert.assertEquals(expected, actual);
     } 
+
+    //remove sessionID from indexName to get fileName
+    @Test
+    public void encodeIndexNameTest() { 
+        String indexName = "file1";
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/IndexNameTest.java:74]: fileName

File: src/main/java/com/google/error_analyzer/backend/IndexName.java (hunk 8 above)
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/IndexName.java:48]: encodeFromStringToHex

File: src/main/java/com/google/error_analyzer/backend/IndexName.java [hunk 12]
```diff
@@ -52,10 +67,24 @@ private static String getSessionId(HttpServletRequest request) throws NullPointe
...
-            logger.fatal("Session ID is null");
+        if(sessionID == "invalidSession") {
+            logger.fatal("Invalid Session");
         }
-        return sessionID.toLowerCase();
+        return sessionID;
     }
 
+    /*returns decoded index name*/
+    private static String decodeIndexName(String indexName)
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/IndexName.java:77]: decodeFromHexToString

File: src/main/java/com/google/error_analyzer/data/constant/FileConstants.java (hunk 4 above)
Comment (shudhatmajain) [reply src/main/java/com/google/error_analyzer/data/constant/FileConstants.java:17]: Ideally, we should sync StoreUrlLogs to master.
Then we should sync encode to StoreUrlLogs.

File: src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java [hunk 13]
```diff
@@ -84,46 +86,48 @@ public void checkAndStoreFileLog_logExceptionCase() {
...
+            .getJsonStringById(indexName, "7");
         String expected = null;
         assertEquals(expected, actual);
     }
 
     /*store the url logs maximum 5 lines in a single API call*/
     @Test
-    public void storeFileAndUrlLog_forUrl() throws  IOException {
-        String fileName = "file1";
+    public void storeFileAndUrlLog_forUrl() throws  IOException,
```
Comment (nishantnahata) [inline src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java:113]: is there a test for invalid url?
Or is it handled at frontend?

File: src/test/java/com/google/error_analyzer/backend/IndexNameTest.java [hunk 14]
```diff
@@ -50,21 +52,32 @@ public void setUp() {
...
-        String expected = "abcdfile1";
+        String expected = "6162636466696c6531";
         String actual = IndexName.getIndexName(request,fileName);
         Assert.assertEquals(expected, actual);
     }
 
     //remove sessionID from indexName to get fileName
     @Test
-    public void getFileNameTest() { 
+    public void getFileNameTest() throws
```
Comment (nishantnahata) [inline src/test/java/com/google/error_analyzer/backend/IndexNameTest.java:65]: please fix the test name :)
for other tests as well

File: src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java (hunk 13 above)
Comment (asthaaaa) [reply src/test/java/com/google/error_analyzer/backend/FileAndUrlLogTest.java:113]: It is handled by the frontend

**Pull request merged**